node/engine/messageservice/p2p-message-service/service.go

```diff
@@ -146,51 +151,49 @@ func (ms *P2PMessageService) setupDht(bootPeers []string) {
 	ms.checkError(err)
 	ms.dht = kademliaDHT
 
-	// Setup notifications so we exchange nitro signing addresses when connected
+	// Setup network connection notifications
 	n := &network.NotifyBundle{}
 	n.ConnectedF = func(n network.Network, conn network.Conn) {
 		ms.logger.Debug().Msgf("notification: connected to peer %s", conn.RemotePeer().String())
-		go ms.sendPeerInfo(conn.RemotePeer(), false)
+		peerInfo := basicPeerInfo{Id: conn.RemotePeer()}
+		ms.newPeerInfo <- peerInfo
 	}
 	n.DisconnectedF = func(n network.Network, conn network.Conn) {
 		ms.logger.Debug().Msgf("notification: disconnected from peer: %s", conn.RemotePeer().String())
 	}
-
 	ms.p2pHost.Network().Notify(n)
+	ms.connectBootPeers(bootPeers)
 
-	expectedPeers := len(bootPeers)
-	if expectedPeers > 0 {
-		// Add bootpeers and wait for connections before proceeding
-		ms.addBootPeers(bootPeers)
-		ms.logger.Info().Msgf("waiting for %d bootpeer connections", expectedPeers)
+	err = ms.dht.Bootstrap(ctx) // Sends FIND_NODE queries periodically to populate dht routing table
+	ms.checkError(err)
+
+	// Must wait until dht RoutingTable has an entry before adding custom dht record
+	go func() {
 		ticker := time.NewTicker(BOOTSTRAP_SLEEP_DURATION)
 		for range ticker.C {
-			peers := ms.p2pHost.Network().Peers()
-			actualPeers := len(peers)
-			ms.logger.Debug().Msgf("found peers: %v, expected peers: %d", actualPeers, expectedPeers)
-			for _, peer := range peers {
-				ms.logger.Debug().Msgf("peer info: %v", peer)
-			}
-
-			// Once we've connected to enough peers, stop the ticker
-			if actualPeers >= expectedPeers {
-				ms.logger.Info().Msgf("initial threshold for peer connections has been met")
+			dhtSize := ms.dht.RoutingTable().Size()
+			ms.logger.Info().Msgf("routing table size: %d", dhtSize)
+			if dhtSize > 0 {
+				ms.addScaddrDhtRecord(ctx)
 				ticker.Stop()
-				break
+				close(ms.initComplete)
+				return
 			}
 		}
-
-		ms.addScaddrDhtRecord(ctx)
-	}
-
-	err = ms.dht.Bootstrap(ctx) // Runs periodically to maintain a healthy routing table
-	ms.checkError(err)
+	}()
 
 	ms.logger.Info().Msgf("DHT setup complete")
 }
 
+// EventFeed returns the out chan, and narrows the type so that external consumers may only receive on it.
```

Comment:
```suggestion
// InitComplete returns a chan that gets closed once the message service is initialized.
```

Reply:
Done in https://github.com/statechannels/go-nitro/pull/1515/commits/b0c076b83b6f8f860bfd52ee7243b605ed734afd